src/cli/commands/test/iac-local-execution/parsers/terraform-plan-parser.ts

```diff
@@ -47,8 +63,36 @@ function extractChildModulesResources(
   return extractedChildModuleResources;
 }
 
+function extractResourceChanges(
+  terraformPlanJson: TerraformPlanJson,
+): Array<TerraformPlanResourceChange> {
+  return terraformPlanJson?.resource_changes || [];
```

Comment:
nit: why is this `?` 
```suggestion
  return terraformPlanJson.resource_changes || [];
```

Reply:
being graceful as possible, so any case someone provides a tf plan without `resource_changes` (like we had with the child_modules), then it defaults to an empty array, instead of assuming it's always there.

Reply:
Ah I see, that part is covered by `resource_changes || []`. The `terraformPlanJson?` is checking whether `terraformPlanJson` is `null` or `undefined` which is not currently allowed via the signature and also feels like it would be an invariant?

Reply:
woops, you're correct.
my brain mistakenly interpreted it as optional chaining for the `resources_changes` field